### PR TITLE
Typescript documentation

### LINKED ISSUE #206: Typescript typings support
I am currently writing typings for react-sketchapp. I plan on publishing these typings to `@types/react-sketchapp` but an alternative to that is having the typings included in this repo itself. [This is done by including a `typings` field in the `package.json` that references a typings file (e.g. `react-sketchapp.d.ts`)](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package).

Including the typings first-party enables better typescript support as well as better javascript support (without a typescript dependency) for editors that use the typescript language service.

Proposal:

If you give me your blessing, I'd like to fully type this library and include those typings in a pull, else it's going to `@types/react-sketchapp`.

### PR DESCRIPTION
**Pull request by ricokahler**
This is a pull to add documentation and an example for using typescript with react-sketchapp. I added typings to DefinitelyTyped and I tested out the example I added. I couldn't get the gitbooks documentation working correctly but the markdown pages I added to `docs` look fine.

Try the example by running:

```bash
curl https://codeload.github.com/ricokahler/react-sketchapp/tar.gz/typescript | tar -xz --strip=2 react-sketchapp-typescript/examples/basic-setup-typescript
cd basic-setup-typescript
npm install
npm run typescript:once
npm start
```

Let me know what you think.

fix #206

I think the only outstanding thing is getting the gitbooks docs to reflect the changes to the markdown files. I'm unfamiliar with gitbooks.

### CODE REVIEW

**Comment by jongold** [comment]
This is awesome, thank you! Ready to merge?

**Comment by ricokahler** [comment]
@jongold yeah if it looks good to you. thanks man

File: examples/basic-setup-typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "typescript:once": "./node_modules/.bin/tsc",
+    "typescript:clean": "rm -rf ./.ts-compiled"
+  },
+  "author": "Jon Gold <jon.gold@airbnb.com>",
+  "license": "MIT",
+  "devDependencies": {
+    "@types/chroma-js": "^1.3.3",
+    "@types/react": "^16.0.25",
+    "@types/react-sketchapp": "^0.12.0",
+    "skpm": "^0.10.2",
```
Comment (mathieudutour) [inline examples/basic-setup-typescript/package.json]: probably should update to `@skpm/builder` (see #203 )

Comment (ricokahler) [reply examples/basic-setup-typescript/package.json]: Is that a one-line change?

Comment (mathieudutour) [reply examples/basic-setup-typescript/package.json]: no there are some scripts to change as well: https://github.com/airbnb/react-sketchapp/pull/203/files#diff-3bcd169ea0fcdf0ddf5666461f84f003

Comment (ricokahler) [reply examples/basic-setup-typescript/package.json]: Alright 👍 I'll update once I have more time

**Comment by ricokahler** [comment]
@jongold @mathieudutour I finally finally updated this (it's been a busy semester).

Try the example and let me know what you think. Should be a simple update.

```
curl https://codeload.github.com/ricokahler/react-sketchapp/tar.gz/typescript | tar -xz --strip=2 react-sketchapp-typescript/examples/basic-setup-typescript
cd basic-setup-typescript
npm install
npm run render
```

**Pull request merged**